src/ansys/openapi/common/_session.py

```diff
@@ -74,6 +81,28 @@
 Api_Client_Factory = TypeVar("Api_Client_Factory", bound="ApiClientFactory")
 
 
+class AuthenticationScheme(Enum):
+    """Authentication scheme.
+
+    Used to specify an authentication scheme used when connecting to the server.
+    """
+
+    AUTO = "auto"
+    """Use the 401 response www-authenticate header to select the best supported authentication scheme."""
+
+    BASIC = "Basic"
+    """Force the use of Basic authentication, even if more secure options are supported."""
+
+    NTLM = "NTLM"
+    """Force the use of NTLM authentication only. Do not fall back to Basic authentication."""
```

Comment:
Writing these docstrings is what makes me think we should converge on AUTO, BASIC, WINDOWS instead of being more granular. Ultimately, you're going to want to specify basic even though you know it's not secure for reasons, or you're going to specify "not basic" because you don't want to use basic for security purposes.

Since the libraries we use for auth don't let us be more specific than "negotiate" in most situations anyway, combining negotiate, ntlm, and kerberos would make the situation a lot simpler.

Reply:
Thinking this through - it probably makes sense to allow the user to specify that they do really want to do NTLM, since you provide credentials. However I dont think it makes sense to allow users to specify Negotiate or Kereberos since those are not credential based auth modes?

Reply:
I actually don't think it will work for negotiate at least, since requests-negotiate-sspi will just error if we don't get 401 Negotiate in the intial response. This feels like a reasonable attitude to take...

Reply:
As discussed, I've removed the new arg for `with_autologon`, and now only expose this for `with_credentials`. Removed the autologon-specific enum values as well, and removed any tests that are no longer relevant.